### ISSUE TITLE
`if [ -z "${#ARRAY[@]}" ]; then` is a no-op check

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#2630]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] ~~I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related~~ I didn't do it, I'm sowwwy :sweat: 

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
set -euo pipefail

mapfile -d '' ARRAY < <(git ls-files -z '*.py')
# If there are no python files we are done
if [ -z "${#ARRAY[@]}" ]; then
  exit 0
fi
```

#### Here's what shellcheck currently says:

_Nothing_

#### Here's what I wanted or expected to see:

`test -z` / `test -n` are "null and void" comparisons for "${#ARRAY[@]}".

Did you mean: `[ "${#ARRAY[@]}"  -eq 0 ]` / `[ "${#ARRAY[@]}"  -ne 0 ]`?
